src/reference/asciidoc/xmpp.adoc

```diff
@@ -80,6 +80,42 @@ When started it will register a `PacketListener` that will listen for incoming X
 It forwards any received messages to the underlying adapter which will convert them to Spring Integration Messages and send them to the specified `channel`.
 It will unregister the `PacketListener` when it is stopped.
 
+Starting with _version 4.3_ the `ChatMessageListeningEndpoint` (and its `<int-xmpp:inbound-channel-adapter>`)
+supports a `org.jivesoftware.smack.filter.StanzaFilter` injection to be registered on the provided `XMPPConnection`
+together with an internal `StanzaListener` implementation.
+See their https://www.igniterealtime.org/builds/smack/docs/latest/javadoc/org/jivesoftware/smack/XMPPConnection.html#addAsyncStanzaListener%28org.jivesoftware.smack.StanzaListener,%20org.jivesoftware.smack.filter.StanzaFilter%29[JavaDocs] for more information.
+
+Also with the _version 4.3_ the `payload-expression` has been introduced for the `ChatMessageListeningEndpoint`.
+The incoming `org.jivesoftware.smack.packet.Message` represents a root object of evaluation context.
+This option is useful in case of <<xmpp-extensions>>.
+For example, for the GCM protocol we can extract the body using expression:
+
+[source,xml]
+----
+payload-expression="getExtension('google:mobile:data').json"
+----
+
+for the XHTML protocol:
+
+[source,xml]
+----
+payload-expression="getExtension(T(org.jivesoftware.smackx.xhtmlim.packet.XHTMLExtension).NAMESPACE).bodies[0]"
+----
+
+To simplify the access to the Extension in the XMPP Message, the `extension` variable is added into the
+`EvaluationContext`.
+Note, it is done only when one and only one Extension is present in the Message.
+The samples above with the `namespace` manipulations can be simplified to something like:
+
+[source,xml]
+ ----
+ payload-expression="#extension.json"
```

Comment:
Hello, @puel !

Thank you for the review.
Very appreciate!

Sorry, I don't understand your concern about "it to the XML documentation of the payload-expression". See the `spring-integration-xmpp-4.3.xsd` changes. There is really a documentation on the `payload-expression` attribute.

If you are not familiar with SpEL, you should go to the Spring Framework [Reference Manual](http://docs.spring.io/spring/docs/current/spring-framework-reference/html/expressions.html).

The same is for the `new SpelExpressionParser()`, just because that part is really outside of Spring Integration scope. That's really a big truth, that to use Spring Integration properly you should know many other Spring projects.

From other hand that's a bit complicated to add explanations for related technologies in each Spring Integration paragraph.

Also see http://docs.spring.io/spring-integration/reference/html/spel.html.